frontend/device/battery.lua

```diff
@@ -0,0 +1,119 @@
+local ButtonDialogTitle = require("ui/widget/buttondialogtitle")
+local Device = require("device")
+local Font = require("ui/font")
+local InputContainer = require("ui/widget/container/inputcontainer")
+local Screen = Device.screen
+local UIManager = require("ui/uimanager")
+local powerd = Device:getPowerDevice()
+local _ = require("gettext")
+local T = require("ffi/util").template
+
+local Battery = InputContainer:new{
+}
+
+function Battery:init()
+    self:scheduleBatteryLevel()
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function Battery:addToMainMenu(menu_items)
+    if powerd:getCapacity() > 0 or powerd:isCharging() then
```

Comment:
I guess you could check that in Battery:init(), as it won't ever change, and save it in self.is_battery_watchable or something